Formula/adios2.rb

```diff
@@ -0,0 +1,81 @@
+class Adios2 < Formula
+  desc "Next generation of ADIOS developed in the Exascale Computing Program"
+  homepage "https://adios2.readthedocs.io"
+  url "https://github.com/ornladios/ADIOS2/archive/v2.5.0.tar.gz"
+  sha256 "7c8ff3bf5441dd662806df9650c56a669359cb0185ea232ecb3578de7b065329"
+  head "https://github.com/ornladios/ADIOS2.git", :branch => "master"
+
+  depends_on "cmake" => :build
+  depends_on "gcc" => :build
+  depends_on "c-blosc"
+  depends_on "libfabric"
+  depends_on "libpng"
+  depends_on "numpy"
+  depends_on "open-mpi"
+  depends_on "python"
+  depends_on "zeromq"
+  uses_from_macos "bzip2"
+
+  resource "mpi4py" do
+    url "https://files.pythonhosted.org/packages/04/f5/a615603ce4ab7f40b65dba63759455e3da610d9a155d4d4cece1d8fd6706/mpi4py-3.0.2.tar.gz"
+    sha256 "f8d629d1e3e3b7b89cb99d0e3bc5505e76cc42089829807950d5c56606ed48e0"
+  end
+
+  def install
+    resource("mpi4py").stage do
+      system "#{Formula["python"].opt_bin}/python3",
+             *Language::Python.setup_install_args(libexec)
+    end
+    xy = Language::Python.major_minor_version "#{Formula["python"].opt_bin}/python3"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
+
+    args = std_cmake_args + %W[
+      -DADIOS2_USE_Blosc=ON
+      -DADIOS2_USE_BZip2=ON
+      -DADIOS2_USE_DataSpaces=OFF
+      -DADIOS2_USE_Fortran=ON
+      -DADIOS2_USE_HDF5=OFF
+      -DADIOS2_USE_MGARD=OFF
+      -DADIOS2_USE_MPI=ON
+      -DADIOS2_USE_PNG=ON
+      -DADIOS2_USE_Python=ON
+      -DADIOS2_USE_SZ=OFF
+      -DADIOS2_USE_ZeroMQ=ON
+      -DADIOS2_USE_ZFP=OFF
+      -DCMAKE_DISABLE_FIND_PACKAGE_BISON=TRUE
+      -DCMAKE_DISABLE_FIND_PACKAGE_CrayDRC=TRUE
+      -DCMAKE_DISABLE_FIND_PACKAGE_FLEX=TRUE
+      -DCMAKE_DISABLE_FIND_PACKAGE_LibFFI=TRUE
+      -DCMAKE_DISABLE_FIND_PACKAGE_NVSTREAM=TRUE
+      -DPYTHON_EXECUTABLE:FILEPATH=#{Formula["python"].opt_bin}/python3
+      -DADIOS2_BUILD_TESTING=OFF
+      -DADIOS2_BUILD_EXAMPLES=OFF
+    ]
+    mkdir "build" do
+      system "cmake", "..", *args
+      system "make", "install"
+      rm_rf Dir[prefix/"bin/bp4dbg"] # https://github.com/ornladios/ADIOS2/pull/1846
+    end
+
+    (pkgshare/"test").install "examples/hello/bpWriter/helloBPWriter.cpp"
+    (pkgshare/"test").install "examples/hello/bpWriter/helloBPWriter.py"
+  end
+
+  test do
+    adios2_config_flags = `adios2-config --cxx`.chomp.split
+    system "mpic++",
+           (pkgshare/"test/helloBPWriter.cpp"),
+           *adios2_config_flags
+    system "./a.out"
+    assert_predicate testpath/"myVector_cpp.bp", :exist?
+
+    system "#{Formula["python"].opt_bin}/python3",
+           "-c", "import adios2"
+    system "#{Formula["python"].opt_bin}/python3",
+           "-c", "import mpi4py"
```

Comment:
These tests look great! I'm not sure we need to test mpi4py import functionality but I'm happy enough to leave it as it is.

Reply:
yes, have this intentionally right now because the `import mpi4py` line does not yet work.